core/auth/fake/interfaces/controller_test.go

```diff
@@ -0,0 +1,397 @@
+package interfaces
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+
+	"flamingo.me/flamingo/v3/framework/web"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type (
+	mockRouter struct {
+		mock.Mock
+		broker string
+	}
+)
+
+var _ web.ReverseRouter = (*mockRouter)(nil)
+
+// Relative mock action
+func (m *mockRouter) Relative(to string, params map[string]string) (*url.URL, error) {
+	panic("not implemented")
+}
+
+// Absolut mock action
+func (m *mockRouter) Absolute(r *web.Request, to string, params map[string]string) (*url.URL, error) {
+	resultURL := &url.URL{}
+
+	return resultURL.Parse(strings.ReplaceAll(FakeAuthURL, ":broker", m.broker))
+}
+
+func Test_idpController_Auth(t *testing.T) {
+	t.Parallel()
+
+	type fields struct {
+		config fakeConfig
+	}
+
+	type args struct {
+		r *web.Request
+	}
+
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   web.Result
+	}{
+		{
+			name: "simple controller call with empty template to test fallback template",
+			fields: fields{
+				config: fakeConfig{},
+			},
+			args: args{
+				r: addRequestParameters(web.CreateRequest(
```

Comment:
creating the request this way adds a lot of duplicate code, why not put the boilerplate into `addRequestParameters`, name it `createTestRequest` and pass in the  method and optional form values?